editor/tests/org.fusesource.ide.camel.editor.tests.integration/src/main/java/org/fusesource/ide/camel/editor/integration/preferences/UserLabelsListEditorTestIT.java

```diff
@@ -0,0 +1,84 @@
+package org.fusesource.ide.camel.editor.integration.preferences;
+
+import org.eclipse.jface.preference.ListEditor;
+import org.eclipse.swt.widgets.Shell;
+import org.fusesource.ide.camel.editor.preferences.UserLabelsListEditor;
+import org.fusesource.ide.preferences.PreferenceManager;
+import org.fusesource.ide.preferences.PreferencesConstants;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * 
+ * @author Andrej Podhradsky (apodhrad@redhat.com)
+ *
+ */
+public class UserLabelsListEditorTestIT {
+
+	private static Shell shell;
+	private ListEditor listEditor;
+
+	@BeforeClass
+	public static void createShell() {
+		shell = new Shell();
+	}
+
+	@Before
+	public void createListEditor() {
+		listEditor = new UserLabelsListEditor(PreferencesConstants.EDITOR_USER_LABELS, "", shell);
+		listEditor.setPreferenceStore(PreferenceManager.getInstance().getUnderlyingStorage());
+	}
+
+	@Test
+	public void testValidInputs() {
```

Comment:
I think that this test can be moved to Unit test instead of integration test (faster feedback)